Apps/Playground/Android/app/src/main/cpp/BabylonNativeJNI.cpp

```diff
@@ -31,40 +33,53 @@ extern "C" {
     JNIEXPORT void JNICALL Java_BabylonNative_Wrapper_setTouchInfo(JNIEnv* env, jobject obj, jfloat dx, jfloat dy, jboolean down);
 };
 
-std::unique_ptr<Babylon::AppRuntime> runtime{};
-std::unique_ptr<InputManager::InputBuffer> inputBuffer{};
-std::unique_ptr<Babylon::ScriptLoader> loader{};
+namespace
+{
+    std::unique_ptr<Babylon::AppRuntime> g_runtime{};
+    std::unique_ptr<InputManager::InputBuffer> g_inputBuffer{};
+    std::unique_ptr<Babylon::ScriptLoader> g_scriptLoader{};
+}
 
-AAssetManager *g_assetMgrNative = nullptr;
+// TODO : this is a workaround for asset loading on Android. To remove when the plugin system is in place.
+AAssetManager* g_assetManager = nullptr;
 
 JNIEXPORT void JNICALL
-Java_BabylonNative_Wrapper_initEngine(JNIEnv* env, jobject obj,
-                                      jobject assetMgr, jobject appContext)
+Java_BabylonNative_Wrapper_initEngine(JNIEnv* env, jobject obj, jobject assetManager)
 {
-    auto asset_manager = AAssetManager_fromJava(env, assetMgr);
-    g_assetMgrNative = asset_manager;
+    // TODO : this is a workaround for asset loading on Android. To remove when the plugin system is in place.
+    g_assetManager = AAssetManager_fromJava(env, assetManager);
 }
 
 JNIEXPORT void JNICALL
 Java_BabylonNative_Wrapper_finishEngine(JNIEnv* env, jobject obj)
 {
-    loader.reset();
-    inputBuffer.reset();
-    runtime.reset();
+    g_scriptLoader.reset();
+    g_inputBuffer.reset();
+    g_runtime.reset();
     Babylon::DeinitializeGraphics();
 }
 
 JNIEXPORT void JNICALL
-Java_BabylonNative_Wrapper_surfaceCreated(JNIEnv* env, jobject obj, jobject surface)
+Java_BabylonNative_Wrapper_surfaceCreated(JNIEnv* env, jobject obj, jobject surface, jobject appContext)
 {
-    if (!runtime)
+    if (!g_runtime)
     {
-        runtime = std::make_unique<Babylon::AppRuntime>("");
+        g_runtime = std::make_unique<Babylon::AppRuntime>("");
 
-        ANativeWindow *window = ANativeWindow_fromSurface(env, surface);
+        JavaVM* javaVM{};
+        if (env->GetJavaVM(&javaVM) != JNI_OK)
+        {
+            throw std::runtime_error("Failed to get Java VM");
+        }
+
+        // TODO: This should be cleaned up via env->DeleteGlobalRef
```

Comment:
should we fix this before merge?

Reply:
My vote is no. I want to stop having to merge cmakelists.txt files :)

Reply:
Okay, maybe we can file an issue on it.